processing/src/main/java/io/druid/query/extraction/AbstractExtractionFn.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.extraction;
+
+import io.druid.segment.column.ValueAccessor;
+
+/**
+ */
+public abstract class AbstractExtractionFn implements ExtractionFn
```

Comment:
making it abstract class is somehow not very user friendly, assume i have my class that already extends another super class related to my own org, by making this abstract class you make it impossible to use it as an extraction function, unless i have to create the object within the class it self. i think we should find a way to make it as an interface.